### ISSUE TITLE
Question About Kernel Space Modules and NVIDIA Tegra K1

### ISSUE THREAD

**Issue by nvidialinuxuser** [OE4T/linux-tegra-5.10#13]
Good morning,

I've recently decided to get an ARM ChromeBook (Acer 13) with an NVIDIA Tegra K1 GPU up and running with Ubuntu Focal (20.04 LTS) + Ubuntu Xenial (16.04 LTS) Xorg packages + NVIDIA L4T 21.8 to get full GPU acceleration.  An issue with this current setup (asides from the old Xorg situation) is that I'm relying on an old Chrome OS kernel which is at version 3.10--I would like to bring up my kernel version to something newer (such as 5.10 LTS).  I did some searching around and found this repo as a result.

Currently, the L4T kernel space provided by NVIDIA in their kernel source are also out of date (3.10).  I looked through this repository and I've seen several references to 'gk20a'.  I noticed the L4T kernel space being used in this repo is significantly newer.  I'm somewhat new to working with Tegra platforms and so I was wondering if the L4T kernel space in this repo is compatible with the Tegra K1 platform.  Hope I'm in the right place to ask this.  Thanks!

**Comment by madisongh** [comment]
No, the only platforms NVIDIA supports with the current L4T kernel are the latest Jetson modules - the Xavier and Orin series.  The 3.10 kernel that came with the R21 series L4T was the last of the NVIDIA downstream kernels to support the TK1.

There is some support for the TK1 and other older Tegra/Jetson platforms in the mainline kernel, although drivers for various Tegra-specific hardware blocks may not be available, and you would use Nouveau for GPU support.  There are some pointers at [elinux.org](https://elinux.org/Tegra/Mainline_SW), although some of the information there may be stale.

**Comment by nvidialinuxuser** [comment]
Thank you for the information!

Nouveau would have been appealing if it weren’t for the fact that most recent Novueau on Mesa versions that do work somewhat need an older Mesa and Xorg to run with a hacky patch that gets it running, though with frequent and severe artifacting, making it difficult to use.  The current and latest Mesa however has several severe unresolved regressions (e.g. most desktop environments don’t function it seems) that appears to have little to no progress fix-wise.

Xenial Xorg + Ubuntu Focal/Jammy appears to work well with minimal artifacting (at least on the Unity desktop) which is a reason why I’ve opted to use NVIDIA’s blobs.  I’ve thought about trying to bring up the L4T 21.8 NVIDIA GPU kernel modules up to some still supported LTS.  

If it’s alright for me to ask, Is this feasible?  Which files that are GPU kernel module-specific that would I need to bring over?

**Comment by madisongh** [comment]
> If it’s alright for me to ask, Is this feasible? Which files that are GPU kernel module-specific that would I need to bring over?

You can ask, but I can't really tell you how feasible this would be.
